android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -718,13 +730,20 @@ public void remove(TiViewProxy child)
 	@Kroll.method
 	public void removeAllChildren()
 	{
-		if (children != null) {
-			//children might be altered while we loop through it (threading)
-			//so we first copy children as it was when asked to remove all children
-			ArrayList<TiViewProxy> childViews = new ArrayList<TiViewProxy>();
-			childViews.addAll(children);
-			for (TiViewProxy child : childViews) {
-				remove(child);
+		if (children != null && view != null) {
+			View nv = view.getNativeView();
+			if (!(nv instanceof TiCompositeLayout) && nv.getClass().getSimpleName()!="ListViewWrapper") {
+				nv = ((ViewGroup) nv).getChildAt(0);
+			}
+				
+			if (nv instanceof ViewGroup) {
+				ViewGroup vg = (ViewGroup) nv;
+				if (TiApplication.isUIThread()) {
+					vg.removeAllViews();
```

Comment:
The original code uses the method ```remove(child)``` to remove the TiViewProxy child view. This in turns call ```handleRemove()```.

In the handleRemove, the TiViewProxy is released by the method ```releaseViews()```. This is not done here. Would there be a leak caused from here?